### ISSUE TITLE
[Abdul] Stage 3: Logistic Regression & Linear SVM (Baselines & Margin Classifiers)

### ISSUE THREAD

**Issue by dipk6545** [dipk6545/hotel-booking-aml#28]
### Assigned Models
1. **Random Forest** (Tree-based ensemble)

### Tasks
- [ ] Train & evaluate baseline Random Forest.
- [ ] Perform Hyperparameter Tuning:
  - `n_estimators`: [50, 100, 200]
  - `max_depth`: [10, 20, None]
  - `class_weight`: [None, 'balanced']
- [ ] Evaluate with 5-Fold Cross-Validation on the training set.
- [ ] Log test set metrics: ROC-AUC, F1-Score, Precision, Recall, Accuracy.
- [ ] Export your best-tuned model configuration for the team stacking ensemble.
